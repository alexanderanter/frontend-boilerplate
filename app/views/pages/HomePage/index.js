import React from 'react';
import { Link } from 'react-router-dom';
import BasePage from '../BasePage';
import FlexBox from '../../common/FlexBox';
import SampleSocket from '../../components/SampleSocket';
import SampleReducer from '../../containers/SampleReducerContainer';

const HomePage = () => (
  <BasePage title="welcome">
    <FlexBox align="center" justify="space-around">
      <p>Sample routes:</p>
      <Link to="/page-one">Page 1</Link>
      <Link to="/page-two">Page 2</Link>
      <Link to="/page-three">Page 3</Link>
    </FlexBox>
    <hr />
    <SampleReducer />
    <SampleSocket />
  </BasePage>
);

export default HomePage;
