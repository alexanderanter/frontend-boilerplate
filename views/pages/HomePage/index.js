import React from 'react';
import { Link } from 'react-router-dom';
import BasePage from '../BasePage';
import FlexBox from '../../common/FlexBox';
import SampleReducer from '../../components/SampleRecuder';

const HomePage = () => (
  <BasePage title="welcome">
    <FlexBox align="center" justify="space-around">
      <p>Sample routes:</p>
      <Link to="/page-one">Page 1</Link>
      <Link to="/page-two">Page 2</Link>
      <Link to="/page-three">Page 3</Link>
    </FlexBox>
    <SampleReducer />
  </BasePage>
);

export default HomePage;
