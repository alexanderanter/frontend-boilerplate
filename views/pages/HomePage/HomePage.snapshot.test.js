import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import HomePage from './index';

describe('HomePage', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<HomePage {...props}>{props.children}</HomePage>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {};
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
