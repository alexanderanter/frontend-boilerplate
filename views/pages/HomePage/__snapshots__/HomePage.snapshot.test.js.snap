// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage without props Should match snapshot 1`] = `
<BasePage
  title="welcome"
>
  <styled.div
    align="center"
    justify="space-around"
  >
    <p>
      Sample routes:
    </p>
    <Link
      replace={false}
      to="/page-one"
    >
      Page 1
    </Link>
    <Link
      replace={false}
      to="/page-two"
    >
      Page 2
    </Link>
    <Link
      replace={false}
      to="/page-three"
    >
      Page 3
    </Link>
  </styled.div>
  <hr />
  <Connect(ReduxForm) />
  <ReduxForm />
</BasePage>
`;
